prime-angular-frontend/src/app/modules/adjudication/shared/components/review-status-content/review-status-content.component.ts

```diff
@@ -126,7 +139,11 @@ export class ReviewStatusContentComponent implements OnInit, OnChanges {
           return reasons.concat(new Reason(this.configPipe.transform(esr.statusReasonCode, 'statusReasons'), esr.reasonNote, this.enrollee.identificationDocuments));
         }
 
-        reasons.push(new Reason(this.configPipe.transform(esr.statusReasonCode, 'statusReasons'), esr.reasonNote));
+
+        reasons.push((esr.statusReasonCode === EnrolmentStatusReasonEnum.POSSIBLE_PAPER_ENROLMENT_MATCH)
```

Comment:
Push is a useful method, but you're making this hard to read when you repurpose the push to contain a ternary.

Reply:
You'll want to pull this apart and to reduce duplication.  For example, `new Reason(this.configPipe.transform(esr.statusReasonCode, 'statusReasons')` occurs 3 times in this method.  Looks like no matter what code path you take you have to invoke it.  Also, `new Reason(this.configPipe.transform(esr.statusReasonCode, 'statusReasons'), esr.reasonNote)` occurs 2 times in the same ternary.

Reply:
Just with those two changes the ternary is now readable:

```ts
if (esr.statusReasonCode === EnrolmentStatusReasonEnum.SELF_DECLARATION) {
  return reasons.concat(this.parseSelfDeclarations(this.enrollee));
}

const statusReason = this.configPipe.transform(esr.statusReasonCode, 'statusReasons');

if (esr.statusReasonCode === EnrolmentStatusReasonEnum.IDENTITY_PROVIDER) {
  return reasons.concat(new Reason(statusReason, esr.reasonNote, this.enrollee.identificationDocuments));
}

const reason = new Reason(statusReason, esr.reasonNote);

reasons.push(
  (esr.statusReasonCode === EnrolmentStatusReasonEnum.POSSIBLE_PAPER_ENROLMENT_MATCH)
    ? this.parsePotentialMatchIds(reason)
    : reason
);

return reasons;
```

Reply:
refactored